src/main/java/com/okta/sdk/clients/UserApiClient.java

```diff
@@ -296,6 +593,109 @@ public LoginCredentials changeRecoveryQuestion(String userId, String currentPass
+    }
+
+    /**
+     * Get all users up to a limit using cursor with paging info.
+     *
+     * @param  after               The cursor that determines which users to return after.
+     * @param  limit               The max number of results returned.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsAfterCursorWithLimit(String after, int limit) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseAfterCursorWithLimit(after, limit));
+    }
+
+    /**
+     * Get all users from url with paging info.
+     *
+     * @param  url                 The URL to get users from.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsByUrl(String url) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseByUrl(url));
+    }
+
+    /**
+     * Get all users using advanced search with paging info.
+     *
+     * @param  filterBuilder       The advanced search filter thats being used to match users.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsWithAdvancedSearch(FilterBuilder filterBuilder) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseWithAdvancedSearch(filterBuilder));
+    }
+
+    /**
+     * Get all users up to a limit using advanced search with paging info.
+     *
+     * @param  filterBuilder       The advanced search filter thats being used to match users.
+     * @param  limit               The max number of results returned.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsWithAdvancedSearchAndLimit(FilterBuilder filterBuilder, int limit) throws IOException {
+        return new PagedResults<User>(getUsersApiResponseWithAdvancedSearchAndLimit(filterBuilder, limit));
+    }
+
+    /**
+     * Get all users up to a limit using advanced search and cursor with paging info.
+     *
+     * @param  filterBuilder       The advanced search filter thats being used to match users.
+     * @param  limit               The max number of results returned.
+     * @param  after               The cursor that determines which users to return after.
+     * @return PagedResults<User>  The users with paging info.
+     * @throws IOException         If an input or output exception occurred.
+     */
+    public PagedResults<User> getUsersPagedResultsWithAdvancedSearchAndLimitAndAfterCursor(FilterBuilder filterBuilder, int limit, String after) throws IOException {
```

Comment:
I'd just like to reiterate a couple design principles (from the API manifesto) to keep in mind with any SDK change:

* API Should Be As Small As Possible But No Smaller
* When in doubt leave it out; you can always add but you can never remove
* Public APIs are forever - one chance to get it right
* Minimize Accessibility of Everything
* Names Matter–the API is a Little Language

Here, the name `getUsersPagedResultsWithAdvancedSearchAndLimitAndAfterCursor()` is very descriptive, but it's almost unwieldy. Also, do we want to call the new functionality "Advanced Search" here or just "Search"? From the client's perspective, there isn't anything "advanced" about it; its just SCIM filter no different than the one it's already using via the `filter` parameter.